scrape/scrape.go

```diff
@@ -1569,18 +1572,39 @@ loop:
 		// Increment added even if there's an error so we correctly report the
 		// number of samples remaining after relabeling.
 		added++
-
+		exemplars = exemplars[:0] // Reset and reuse the exemplar queue.
```

Comment:
queue -> slice (or maybe just remove the comment – it's a pretty common pattern in Go).